### PR TITLE
Add reset sankey button to no flows error message

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: scripts/actions/tool.actions.js [hunk 1]
```diff
@@ -44,6 +44,65 @@ export function resetState(refilter = true) {
...
   };
 }
+
+// Resets sankey's params that may lead to no flows being returned from the API
+export function resetSankey() {
+  return (dispatch, getState) => {
+    const state = getState();
+    const context = state.tool.contexts.find(context => context.id === state.tool.selectedContextId);
+    const defaultColumns = state.tool.columns.filter(column => column.isDefault);
+    const defaultResizeBy = context.resizeBy.find(resizeBy => resizeBy.isDefault);
```
Comment (sorodrigo) [inline scripts/actions/tool.actions.js]: this is an unsafe access. I would add a fallback `context && context.resizeBy`

File: scripts/components/tool/tool-content.component.js [hunk 2]
```diff
@@ -1,21 +1,27 @@
-
 export default class {
 
   onCreated() {
     this._setVars();
+
+    this.onSankeyReset = this._resetSankey.bind(this);
```
Comment (sorodrigo) [inline scripts/components/tool/tool-content.component.js:6]: It feels weird to bind in `onCreated` instead that in the `constructor`.

File: scripts/components/tool/tool-content.component.js [hunk 3]
```diff
@@ -36,5 +42,11 @@ export default class {
 
   toggleError(hasError) {
     this.sankeyError.classList.toggle('is-hidden', !hasError);
+
+    if (hasError) {
```
Comment (sorodrigo) [inline scripts/components/tool/tool-content.component.js]: I think we can add the event listener on constructor, doesn't make sense to toggle add/remove depending on `hasError`

File: scripts/actions/tool.actions.js (hunk 1 above)
Comment (tiagojsag) [reply scripts/actions/tool.actions.js]: true, but handling a case where a context is missing would be a totally different beast, so I'm adding that validation, but not really handling the case where context is missing

File: scripts/components/tool/tool-content.component.js (hunk 2 above)
Comment (tiagojsag) [reply scripts/components/tool/tool-content.component.js:6]: We're using the same pattern in other places, I'm just reusing it

File: scripts/components/tool/tool-content.component.js (hunk 3 above)
Comment (tiagojsag) [reply scripts/components/tool/tool-content.component.js]: moved

**Pull request merged**